kedro/framework/cli/starters.py

```diff
@@ -91,23 +162,20 @@ def new(
             "Cannot use the --directory flag without a --starter value."
         )
 
-    # The `astro-iris` was renamed to `astro-airflow-iris`, but old (external) documentation
-    # and tutorials still refer to `astro-iris`. The below line checks if a user has entered old
-    # `astro-iris` as the starter name and changes it to `astro-airflow-iris`.
-    starter_name = (
-        "astro-airflow-iris" if starter_name == "astro-iris" else starter_name
-    )
-    if starter_name in _STARTER_ALIASES:
+    starter_aliases = _get_starters_aliases()
+
+    if starter_name in starter_aliases:
         if directory:
             raise KedroCliError(
                 "Cannot use the --directory flag with a --starter alias."
             )
-        template_path = _STARTERS_REPO
-        directory = starter_name
-        checkout = checkout or version
+        template_path = starter_aliases[starter_name]["template_path"]
+        # "directory" is an optional key for starters from plugins, so if the key is not present we will use "None".
+        directory = starter_aliases[starter_name].get("directory")
+        checkout = checkout or starter_aliases[starter_name].get("checkout", version)
```

Comment:
I agree with @AntonyMilneQB that we leave `checkout` out of the scope for now, and assume the user-defined starters follow the same tagging/branching convention (default to sth that matches the kedro version). This is currently what happens anyway for custom starters, and I think it's a more complicated discussion to have than we could do justice for here.

Reply:
Yes, sure. I did not add checkout in the first version and it really seems that it needs to be discussed further.